plugin.rb

```diff
@@ -16,5 +16,58 @@ module ::MyPluginModule
 require_relative "lib/my_plugin_module/engine"
 
 after_initialize do
-  # Code which should run after Rails has finished booting
+  reloadable_patch do
+    module ::TopicControllerNoIndexExtension
+      Discourse::Application.routes.prepend do
+        put "t/:topic_id/toggle-noindex" => "topics#toggle_noindex",
+            :constraints => {
+              topic_id: /\d+/,
+            }
+      end
+
+      def toggle_noindex
+        head :forbidden and return unless guardian.is_staff?
+        topic_id = params.require(:topic_id).to_i
+        begin
+          topic = Topic.find(topic_id)
+          current = topic.custom_fields["noindex"]
+          newval = nil
+          if current.nil? || current == "f"
```

Comment:
I think if you add this inside after_initialize

```
register_custom_field_type "noindex", :boolean
```
Then core will handle the type casting automatically, and you can just work with `true` / `false` booleans.